src/util.c

```diff
@@ -415,6 +415,133 @@ int ull2string(char *dst, size_t dstlen, unsigned long long value) {
     return 0;
 }
 
+#ifdef HAVE_AVX512
+#include <immintrin.h>
+
+#define MULTIPLIER_10E8 100000000
+#define MULTIPLIER_10E16 10000000000000000ULL
+
+/**
+ * Convert a string into an signed 64-bit integer using AVX-512 instructions.
+ *
+ * This function parses a string of digits and converts it into an signed
+ * 64-bit integer. It leverages AVX-512 SIMD instructions for optimized
+ * processing and performs strict validation to ensure the input string
+ * represents a valid signed integer.
+ *
+ * Notes:
+ * - The input string must not contain leading zeros unless it is "0".
+ * - The function checks for overflow and returns 0 if the result exceeds
+ *   the maximum value of LLONG_MAX.
+ * - This function requires AVX-512 support and will only be compiled if
+ *   `HAVE_AVX512` is defined.
+ *
+ * Example:
+ * Input: s = "1234567890", slen = 10
+ * Steps:
+ * 1. Load the string into SIMD registers and subtract '0' from each character
+ *    to convert ASCII digits to integers.
+ *    Format (32 bytes, 8 bits each):
+ *      0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 2 3 4 5 6 7 8 9 0
+ * 2. Validate that all characters are digits (0-9).
+ * 3. Multiply by 10 and horizontally add adjacent pairs of intermediate values.
+ *    Format (16 bytes, 8 bits each):
+ *      00 00 00 00 00 00 00 00 00 00 00 12 34 56 78 90
+ * 4. Multiply by 100 and horizontally add adjacent pairs of intermediate values.
+ *    Format (8 bytes, 16 bits each):
+ *      0000 0000 0000 0000 0000 0012 3456 7890
+ * 5. Multiply by 10000 and horizontally add adjacent pairs of intermediate values.
+ *    Format (4 bytes, 32 bits each):
+ *      00000000 00000000 00000012 34567890
+ * 6. Extract the final value.
+ * Output: *value = 1234567890
+ */
+ATTRIBUTE_TARGET_AVX512
+static int string2llAVX512(const char *s, unsigned long slen, long long *value) {
+    const char *p = s;
+    unsigned long plen = 0;
+    int negative = 0;
+
+    /* Abort if length indicates this cannot possibly be an int */
+    if (slen == 0 || slen >= LONG_STR_SIZE) return 0;
+
+    /* Special case: first and only digit is 0. */
+    if (slen == 1 && p[0] == '0') {
+        if (value != NULL) *value = 0;
+        return 1;
+    }
```

Comment:
Could we modify this case to include all single-digit numbers without any performance impact?

In the hot path in command parsing, `string2ll()` is used in `processMultibulkBuffer()`. Here, I guess it's often single digit when command names are parsed and key names are parsed.

```suggestion
    /* Special case: first and only digit is 0. Also handle single-digit 1-9 here. */
    if (slen == 1 && p[0] >= '0' && p[0] <= '9') {
        if (value != NULL) *value = p[0] - '0';
        return 1;
    }
```

Reply:
Shall we do this in `string2llScalar()` too?